tests/rptest/tests/workload_upgrade_runner_test.py

```diff
@@ -0,0 +1,288 @@
+            return None
+
+    def begin(self):
+        self._exec_workload_method(WorkloadAdapter.STARTED, "begin")
+
+    def end(self):
+        self._exec_workload_method(WorkloadAdapter.STOPPED, "end")
+
+    def partial_progress(self, versions: dict[Any, RedpandaVersionTriple]):
+        res = self._exec_workload_method(self.state, "partial_progress",
+                                         versions)
+        return res if res is not None else PWorkload.DONE
+
+    def get_workload_name(self):
+        return self.workload.get_workload_name()
+
+    def progress(self, version: RedpandaVersionTriple):
+        res = self._exec_workload_method(self.state, "progress", version)
+        return res if res is not None else PWorkload.DONE
+
+
+class RedpandaUpgradeTest(PreallocNodesTest):
+    def __init__(self, test_context):
+        # si_settings are needed for LicenseWorkload
+        super().__init__(test_context=test_context,
+                         num_brokers=3,
+                         si_settings=SISettings(test_context),
+                         node_prealloc_count=1)
+        # it is expected that older versions of redpanda will generate this kind of errors, at least while we keep testing from v23.x
+        self.redpanda.si_settings.set_expected_damage({
+            'ntr_no_topic_manifest',
+            'ntpr_no_manifest',
+            'unknown_keys',
+            'missing_segments',
+        })
+
+        self.installer = self.redpanda._installer
+
+        # workloads that will be excuted during this test
+        workloads: list[PWorkload] = [
+            DummyWorkload(self),
+            MinimalWorkload(self),
+            LicenseWorkload(self),
+            ProducerConsumerWorkload(self),
+        ]
+
+        # setup self as context for the workloads
+        self.adapted_workloads: list[WorkloadAdapter] = [
+            WorkloadAdapter(workload=w, ctx=self, installer=self.installer)
+            for w in workloads
+        ]
+
+        self.upgrade_steps: list[RedpandaVersionTriple] = []
+
+    def setUp(self):
+        # at the end of setUp, self.upgrade_steps will look like this:
+        # [(22, 1, 11), (22, 1, 10), (22, 1, 11),
+        #  (22, 2, 11), (22, 2, 10), (22, 2, 11),
+        #  (22, 3, 16), (22, 3, 15), (22, 3, 16),
+        #  (23, 1, 7), (23, 1, 6), (23, 1, 7),
+        #  (23, 2, 0)]
+
+        # compute the upgrade steps, merging the upgrade steps of each workload
+        workloads_steps = [
+            self.installer.release_range(
+                earliest=w.get_earliest_applicable_release(),
+                latest=w.get_latest_applicable_release())
+            for w in self.adapted_workloads
+        ]
+
+        latest_unsupported_line = self.installer.latest_unsupported_line()
+        # keeping only releases older than latest EOL.
+        forward_upgrade_steps = [
+            v for v in sorted(set(sum(workloads_steps, start=[])))
+            if v >= latest_unsupported_line
```

Comment:
Shouldn't this be `v > latest_unsupported_line`. Also doesn't `RedpandaInstaller.release_range` do this filtering already?